src/stan/services/util/create_param_mask.hpp

```diff
@@ -0,0 +1,55 @@
+#ifndef STAN_SERVICES_UTIL_CREATE_PARAM_MASKS
+#define STAN_SERVICES_UTIL_CREATE_PARAM_MASKS
+
+#include <stan/io/var_context.hpp>
+#include <rapidjson/document.h>
+#include <string>
+#include <vector>
+#include <stan/math.hpp>
+
+namespace stan{
+namespace services{
+namespace util{
+
+template<typename T>
+void recurse_obj(std::vector<double>& v, const T& rtype_obj, bool clamped){
+  std::string name = rtype_obj["name"].GetString();
+  if (name == "array") {
+    for(int i = 0; i < rtype_obj["length"].GetInt(); i++){
+      recurse_obj(v, rtype_obj["element_type"].GetObject(), clamped);
+    }
+  }
+  else if(name == "matrix") {
+    v.insert(v.end(), rtype_obj["rows"].GetInt() * rtype_obj["cols"].GetInt(),
+	     clamped ? 1 : 0);
+  }
+  else if(name == "vector") {
+    v.insert(v.end(), rtype_obj["length"].GetInt(), clamped ? 1 : 0);
+  }
+  else{
+    v.push_back(clamped ? 1 : 0);
+  }
+}
+
+Eigen::VectorXd create_param_mask(std::string& unconstrained_param_str, stan::io::var_context& clamped){
```

Comment:
Looks good. Put comments on these.

```
/**
 * Create a parameter mask from json encoded dimension information and a var_context which contains the
 * masked variables
 *
 * @param unconstrained_param_str json string
 * @param clamped var_context with clamped parameters
 * @return mask as Eigen::VectorXd with 1s for unclamped parameters and 0s for clamped parameters
 */
```

Or something like that. If it seems hard to write, just do something simpler.